src/actionCreators/resources.js

```diff
@@ -113,27 +128,38 @@ const stubResource = useDefaults => (dispatch, getState) => {
  *
  * Private, but exporting for testing.
  *
- * If the resource template is available in state, it should be provided in the existingResourceTemplate parameter.
- * Otherwise, it will be fetched.
+ * Any resource templates that are needed by not in provided resource templates will be retrieved
+ * and added to the returned resources templates.
+ *
+ * If there is an error retrieving a resource template, undefined will be returned.
  *
  * @param {string} resourceTemplateId
- * @param {Object} existingResourceTemplate the resource template retrieved from state or undefined
+ * @param {Object} resourceTemplates resource templates indexed by resource template id
  * @param {Object} resource existing values for the resource or {}
  * @param {Array<String>} reduxPath of the resource (ending in the resource key or "resource")
  * @param {boolean} useDefaults if true, use default values for property refs
  * @param {boolean} stubMandatoryOnly if true, only stub properties that are mandatory
  * @param {string} stubPropertyURIOnly limit stubbing to this single property or undefined
  * @param {func} dispatch
- * @return {Promise<Object>} the stubbed resource
+ * @return {Promise<[Object, Object]>} the stubbed resource, resource templates
  */
-export const stubResourceProperties = async (resourceTemplateId, existingResourceTemplate,
+export const stubResourceProperties = async (resourceTemplateId, resourceTemplates,
   resource, reduxPath, useDefaults, stubMandatoryOnly, stubPropertyURIOnly, dispatch) => {
   const newResource = _.cloneDeep(resource)
   const newResourceReduxPath = [...reduxPath, resourceTemplateId]
-  const resourceTemplate = existingResourceTemplate || await fetchResourceTemplate(resourceTemplateId, dispatch)
+  let newResourceTemplates = _.cloneDeep(resourceTemplates)
+  let resourceTemplate
+  if (newResourceTemplates[resourceTemplateId]) {
+    resourceTemplate = newResourceTemplates[resourceTemplateId]
+  } else {
+    resourceTemplate = await fetchResourceTemplate(resourceTemplateId, dispatch)
+    newResourceTemplates[resourceTemplateId] = resourceTemplate
+  }
+
   // This handles if there was an error fetching resource template
   if (!resourceTemplate) {
-    return newResource
+    console.error(`Unable to load ${resourceTemplateId}`)
```

Comment:
Could this message need to be propagated up to the UI to alert the user that a resource template could not be fetched?

Reply:
It already is, in the error handling in this function: https://github.com/LD4P/sinopia_editor/blob/master/src/actionCreators/resourceTemplates.js#L12

This is just downstream.